app/middleware/continuation-locals.js

```diff
@@ -0,0 +1,18 @@
+const cls = require('continuation-local-storage');
+
+
+module.exports = (app, conf, log) => {
+  log.info('adding continuation local storage');
+
+  const ns = cls.createNamespace(conf.continuation_locals.namespace);
+
+  return (req, res, next) => {
+    ns.bindEmitter(req);
+    ns.bindEmitter(res);
+
+    ns.run(() => {
```

Comment:
Is this test code or needs to be done on each request?

Reply:
This is what associates the context with the function call chain

Reply:
Do you need to call `ns.set('test', 'foo')` in order for this to happen? If so it may be worth adding a comment and/or change the key/value to something more revealing.

Reply:
Sorry, didn't see what you were talking about. I wil remove that.